gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/login/LoginEndpoint.java

```diff
@@ -107,48 +112,71 @@ private void prepareContext(RoutingContext routingContext, Client client) {
         Map<String, String> params = new HashMap<>(evaluableRequest.getParams().toSingleValueMap());
         params.put(ConstantKeys.ERROR_PARAM_KEY, error);
         params.put(ConstantKeys.ERROR_DESCRIPTION_PARAM_KEY, errorDescription);
+        if (!isTwoStepLoginEnabled) {
+            params.remove(USERNAME_PARAM_KEY);
```

Comment:
I need to check whether if `twoStepLoginEnabled == true and isNullOrEmpty(username)`